browser/brave_shields/cookie_expiry_browsertest.cc

```diff
@@ -0,0 +1,228 @@
+    EXPECT_LE(
+        (base::Time::Now() + less_than_max - cookie.ExpiryDate()).magnitude(),
+        kMarginForTesting);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForCookieStoreMoreThanMax) {
+  ASSERT_TRUE(https_server_->Start());
+
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  JSCookieStoreWriteCookie(browser(),
+                           std::to_string(k4YearsInDays.InMilliseconds()));
+
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_EQ((cookie.ExpiryDate() - cookie.CreationDate()).InDays(), 7);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForHttpCookiesLessThanMax) {
+  auto less_than_max = base::Days(30);
+  std::string cookie_string =
+      "test=http;max-age=" + std::to_string(less_than_max.InSeconds());
+  https_server_->RegisterRequestHandler(base::BindLambdaForTesting(
+      [&](const net::test_server::HttpRequest& request)
+          -> std::unique_ptr<net::test_server::HttpResponse> {
+        auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+        response->AddCustomHeader("Set-Cookie", cookie_string);
+        return std::move(response);
+      }));
+  ASSERT_TRUE(https_server_->Start());
+
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_LE(
+        (base::Time::Now() + less_than_max - cookie.ExpiryDate()).magnitude(),
+        kMarginForTesting);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest,
+                       CheckExpiryForHttpCookiesMoreThanMax) {
+  std::string cookie_string =
+      "test=http;max-age=" + std::to_string(k4YearsInDays.InSeconds());
+  https_server_->RegisterRequestHandler(base::BindLambdaForTesting(
+      [&](const net::test_server::HttpRequest& request)
+          -> std::unique_ptr<net::test_server::HttpResponse> {
+        auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+        response->AddCustomHeader("Set-Cookie", cookie_string);
+        return std::move(response);
+      }));
+  ASSERT_TRUE(https_server_->Start());
+
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_EQ((cookie.ExpiryDate() - cookie.CreationDate()).InDays(), 180);
+  }
+}
```

Comment:
@ShivanKaul did you do a sanity check to make sure that the real time forwarding on a machine actually prevents the cookie to be returned to a website?
I might be a little paranoid, but if our changes are in the wrong place, then `ExpiryDate` can say right things, but the real scenario might fail.

Reply:
Yep!